docs/pages/components/form.md

```diff
@@ -851,3 +851,239 @@ Do not use the checkbox control if:
+                </div>
+                <div class="fd-form-edit-mode fd-form-edit-mode--6">
+                    <input class="fd-input" type="text" id="input-2cc" placeholder="Field placeholder text">
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
+{% endcapture %}
+{% include display-component.html component=form-edit %}
+
+<br>
+
+## Form with 4-7-1 compact Layout
+
+{% capture form-edit %}
+<div class="fd-edit-mode-container">
+    <div class="fd-form-edit-mode-row fd-form-edit-mode-row--compact">
+        <div class="fd-form-edit-mode fd-form-edit-mode--4 fd-form-edit-mode--label">
+            <div class="fd-form-edit-mode-flex fd-form-edit-mode-flex--right">
+                <label class="fd-form-label" for="input-1">Default Input:</label>
+            </div>
+        </div>
+        <div class="fd-form-edit-mode fd-form-edit-mode--7">
+            <input class="fd-input fd-input--compact" type="text" id="input-1" placeholder="Field placeholder text">
+        </div>
+    </div>
+    <div class="fd-form-edit-mode-row fd-form-edit-mode-row--compact">
+        <div class="fd-form-edit-mode fd-form-edit-mode--4 fd-form-edit-mode--label">
+            <div class="fd-form-edit-mode-flex fd-form-edit-mode-flex--right">
+                <label class="fd-form-label fd-form-label--required" for="input-1c">Required Input:</label>
+            </div>
+        </div>
+        <div class="fd-form-edit-mode fd-form-edit-mode--7">
+            <input class="fd-input fd-input--compact" type="text" id="input-1" placeholder="Field placeholder text">
+        </div>
+    </div>
+    <div class="fd-form-edit-mode-row fd-form-edit-mode-row--compact">
+        <div class="fd-form-edit-mode fd-form-edit-mode--4 fd-form-edit-mode--label">
+            <div class="fd-form-edit-mode-flex fd-form-edit-mode-flex--right">
+                <label class="fd-form-label" for="input-2">2 Inputs:</label>
+            </div>
+        </div>
+        <div class="fd-form-edit-mode fd-form-edit-mode--7">
+            <div class="fd-form-edit-mode-row fd-form-edit-mode-row--compact">
+                    <div class="fd-form-edit-mode fd-form-edit-mode--6">
+                        <input class="fd-input fd-input--compact" type="text" id="input-2" placeholder="Field placeholder text">
+                    </div>
+                    <div class="fd-form-edit-mode fd-form-edit-mode--6">
+                        <input class="fd-input fd-input--compact" type="text" id="input-3" placeholder="Field placeholder text">
+                    </div>
+                </div>
+        </div>
+    </div>
+</div>
+{% endcapture %}
+{% include display-component.html component=form-edit %}
+
+<br>
+
+## Form with 12 column Layout
+
+{% capture form-edit %}
+<div class="fd-edit-mode-container">
+    <div class="fd-form-edit-mode-row fd-form-edit-mode-row--vertical">
```

Comment:
I don't get, why you need to put there markup different than for grids
![image](https://user-images.githubusercontent.com/26483208/82921866-2c38ca80-9f79-11ea-81d6-ccc67328fce1.png)
Look at 2 last examples
This is what I achieved it with grid (Only one thing you need to do is lower label's margin/center it vertically)

Reply:
That's the code
```
<div class="fd-container">
    <div class="fd-row">
        <div class="fd-col fd-col--4" style="text-align: right">
            <label class="fd-form-label" for="grid-input-1">Default Input:</label>
        </div>
        <div class="fd-col fd-col--7">
            <input class="fd-input fd-input--compact" type="text" id="grid-input-1" placeholder="Field placeholder text">
        </div>
    </div>
    <div class="fd-row">
        <div class="fd-col fd-col--4" style="text-align: right">
            <label class="fd-form-label" for="grid-input-1">Default Input:</label>
        </div>
        <div class="fd-col fd-col--7">
            <div class="fd-row">
                <div class="fd-col fd-col--6">
                    <input class="fd-input fd-input--compact" type="text" id="grid-input-1" placeholder="Field placeholder text">
                </div>
                <div class="fd-col fd-col--6">
                    <input class="fd-input fd-input--compact" type="text" id="grid-input-1" placeholder="Field placeholder text">
                </div>
            </div>
        </div>
    </div>
</div>```

Reply:
You are right, ill need to import the layout grid and fix this! BUT will still require addition markup for text alignment

Reply:
Lets just verify with the team it is ok, i had done it this way for self-contained but well see what they say